zkevm-circuits/src/copy_circuit.rs

```diff
@@ -389,23 +405,43 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                 }
             );
 
-            // The address is incremented by 1, except in the front mask because the row address has not caught up with the address of the event yet.
-            cb.condition(not_last_two_rows.expr(),
-                |cb| {
+            let front_mask_next = meta.query_advice(front_mask, Rotation(2));
+            let front_mask = meta.query_advice(front_mask, Rotation::cur());
+            cb.require_boolean("front_mask is boolean", front_mask.expr());
+
+            cb.require_zero("front_mask=1 implies mask=1",
+                and::expr([
+                    front_mask.expr(),
+                    not::expr(meta.query_advice(mask, Rotation::cur())),
+                ]),
+            );
+
+            cb.condition(not::expr(is_word_continue.is_lt(meta, None)), |cb| {
+                // The first 31 bytes may be front_mask, but at least the 32nd byte is not masked.
+                cb.require_zero("front_mask = 0 by the end of the first word", front_mask.expr());
+            });
+
+            cb.condition(is_continue.expr(),
+            |cb| {
 
-                    let addr_diff = not::expr(meta.query_advice(front_mask, Rotation::cur()));
+                    cb.require_boolean("front_mask can go from 1 to 0", front_mask.expr() - front_mask_next);
```

Comment:
need to constrain `front_mask = 0` for  back mask rows?

Reply:
`front_mask` should be 0 when the real length reaches 1 because there must be at least one non-masked byte. I need to refactor the length logic too, not part of this PR.